modules/tensor_mechanics/src/materials/ADLAROMANCEStressUpdateBase.C

```diff
@@ -169,350 +237,334 @@ ADLAROMANCEStressUpdateBase::initQpStatefulProperties()
 }
 
 ADReal
-ADLAROMANCEStressUpdateBase::computeResidual(const ADReal & effective_trial_stress,
-                                             const ADReal & scalar)
+ADLAROMANCEStressUpdateBase::maximumPermissibleValue(const ADReal & effective_trial_stress) const
 {
-  if (_immobile_function)
-    _immobile_old = _immobile_function->value(_t, _q_point[_qp]);
-  else
-    _immobile_old = _immobile_dislocations_old[_qp];
+  // Make maximum allowed scalar a little bit less than the deformation that would reduce the trial
+  // stress to zero. This prevents negative trial stresses.
+  return effective_trial_stress / _three_shear_modulus * 0.999999;
```

Comment:
I agree that this negative trial stress error is quite aggravating (all four of my laromance simulations I'm currently running are complaining with that precise error); however, I am really not a fan of this magic number introduction. Why 6 digits of 9s instead of 5 or 7? Is there a better way to achieve the same goal?

Reply:
I am open to suggestions, what would you propose?

Reply:
@lynnmunday and I talked with @bwspenc today about this negative trial stress error problem and have another solution for this error: Ben has a check and reset approach to make sure that the inelastic effective strain (scalar) does not exceed the total mechanical strain in each time step. I believe that @lynnmunday will push up a commit with this change today

Reply:
Awesome!!

Reply:
So, I guess we still need to find a better resolution to this, but for the time being, I think this is OK.

Reply:
> So, I guess we still need to find a better resolution to this, but for the time being, I think this is OK.

I agree. I'm hopeful that with the checks on the dislocation densities, we could explore the check_ranges option that Lynn had tried again in the next development round